### PR TITLE
feat: add determineDirectIncrement Module

### PR DESCRIPTION
**Pull request by santiph**

### CODE REVIEW

File: new/analyse/determine-direct-increment.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      const affectedPackage = clonePackages[affectedPackageName];
+
+      affectedPackage.determinedIncrementLevel = Math.max(
+        affectedPackage.determinedIncrementLevel,
+        mapLevelToIncrementLevel[commitMessage.level]
+      ) as Package['determinedIncrementLevel'];
+
+      // Track affected packages to be releaed with their increemnt level
+      // just for log purposes
+      const possibleLevels = Object.keys(
```
Comment (matthias-reis) [inline new/analyse/determine-direct-increment.ts:55]: You don't seem to do anything with this.

File: new/analyse/determine-direct-increment.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        mapLevelToIncrementLevel
+      ) as Message['level'][];
+      packagesToRelease[affectedPackageName] =
```
Comment (matthias-reis) [inline new/analyse/determine-direct-increment.ts:58]: You could directly spit out a message here and avoid a second traversing below.

**Review by matthias-reis — CHANGES_REQUESTED** [review]

File: new/analyse/determine-direct-increment.ts (hunk 2 above)
Comment (santiph) [reply new/analyse/determine-direct-increment.ts:58]: good point

Comment (santiph) [reply new/analyse/determine-direct-increment.ts:58]: oh, wait...
Different commits could be touching one same package with multiple levels (for example: `misc` when updating tests, and `minor` when adding a feature later on).

Is it ok to log increments for the same package more than once?

File: new/analyse/determine-direct-increment.ts (hunk 1 above)
Comment (santiph) [reply new/analyse/determine-direct-increment.ts:55]: It's used on line 59 `possibleLevels[affectedPackage.determinedIncrementLevel + 1];`

Comment (matthias-reis) [reply new/analyse/determine-direct-increment.ts:55]: 👍

**Review by matthias-reis — APPROVED** [review]

File: new/analyse/determine-direct-increment.ts (hunk 2 above)
Comment (matthias-reis) [reply new/analyse/determine-direct-increment.ts:58]: Yep. We're working on the log output anyway in the end.

**Pull request merged**